evo_researcher/benchmark/utils.py

```diff
@@ -18,6 +18,30 @@ class Market(BaseModel):
     p_yes: float
     volume: float
     is_resolved: bool
+    resolution: str | None
+    outcomePrices: list[float] | None
+
+    @validator("outcomePrices", pre=True)
+    def _validate_outcome_prices(cls, value: list[float] | None) -> list[float] | None:
+        if value is None:
+            return None
+        if len(value) != 2:
+            raise ValueError("outcomePrices must have exactly 2 elements.")
+        return value
+    
+    @property
+    def p_no(self) -> float:
+        return 1 - self.p_yes
+
+    @property
+    def yes_outcome_price(self) -> float:
+        # Use the outcome price if available, otherwise assume it's p_yes.
+        return self.outcomePrices[0] if self.outcomePrices else self.p_yes
```

Comment:
Is this ever not == `self.p_yes`?

Reply:
Polymarket returns `outcomePrices`, so I thought, why not use it directly? I never seen `self.outcomePrices[0] != self.p_yes` (as we are doing `p_yes=m_json["outcomePrices"][0]` in `get_polymarket_markets`), but I have seen `self.outcomePrices[1] != (1 - p_yes)`, so it does make sense to differ it.